### DISCUSSION TITLE
Consult about API for report/monitoring

### DISCUSSION THREAD

**Discussion by manueljordan** [spring-projects/spring-integration#3575] [Q&A] [answered]
Hello SI team

Just being curious, is the an internal API to export or apply monitoring through JMX about all the Adapters being used in any app using Spring Integration? It to be exposed to either JMX and Spring Boot. Of course the latter using the former.
The goal is get a report with or without Spring Boot but working with JMX

My purpose is get my hands dirty with Spring Integration and I am assuming that Adapters is a good start of point. I thought to start first with javadocs, but creating this new module or feature is going to include the javadoc anyway. I though about to start with Channels and Messaging (Message object) too - but I am assuming that there is nothing to create there

How common is for app and developer know at runtime all the adapters being currently executed?

Furthermore I think that a "kind of status check" would be added to each adapter.

Let me know your thoughts

**Comment by artembilan** [comment, marked as answer]
I'm not fully sure what you mean, but probably whatever we have so far in the project should give us a chance to be on the same page with your expectations.
Please, take a look here: https://docs.spring.io/spring-integration/docs/current/reference/html/system-management.html#system-management-chapter

**Reply by manueljordan** [reply]
Oh I see, practically it already exists. I am going to read it in deep.

The idea is: if the app is using or includes Spring Integration and it uses 10 adapters (http, ftp, jdbc, etc) - thus SI should enable and offer some metrics in real time about the 10 adapters and their status. I hope it be more clear now.

Thanks for your understanding

**Reply by artembilan** [reply]
Correct. That's what we do with those metrics via Micrometer integration.

The JMX view is also enhanced with that info, but in addition it provides really management features like start, stop or pause.

Do you really see that we need to expand existing functionality somehow?

BTW with Spring Cloud Sleuth you also has an ability to trace your messages: https://spring.io/projects/spring-cloud-sleuth

**Reply by manueljordan** [reply]
> Do you really see that we need to expand existing functionality somehow?

Now, not, seems all is covered. 

I am going to check the "issues" reported to see if I am able to fix or improve something :)

**Reply by artembilan** [reply]
Cool. Let's consider then close this discussion as answered!

(Not sure how to do that since I'm new with this Discussion feature)

**Reply by manueljordan** [reply]
Seems is not possible, even by my side.

Thanks for the quick and polite feedback

**Reply by garyrussell** [reply]
If you change it to the `Q&A` category (top right), you can mark it as answered.
